### PR TITLE
Remove code for pre-ember-v4

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
Ember v4 is a requirement for the next release of `@ember/test-helpers`, so we can drop some code branches.

### CODE REVIEW

**Comment by NullVoxPopuli** [comment]
Something seems wrong with `@glimmer/manager` :(

**Comment by MelSumner** [comment]
Talked with @NullVoxPopuli and think this could also be good to have in a 3.x release.

File: addon/addon-test-support/@ember/test-helpers/setup-rendering-context.ts [hunk 1]
```diff
@@ -129,55 +128,33 @@ export function render(
       let OutletTemplate = lookupOutletTemplate(owner);
       let ownerToRenderFrom = options?.owner || owner;
 
-      if (macroCondition(dependencySatisfies('ember-source', '<3.24.0'))) {
```
Comment (NullVoxPopuli) [inline addon/addon-test-support/@ember/test-helpers/setup-rendering-context.ts:132]: this code change is the removal of this if branch, keeping only the else

File: addon/addon-test-support/@ember/test-helpers/-internal/is-component.ts [hunk 2]
```diff
@@ -15,15 +14,21 @@ function isComponent(
   maybeComponent: object,
   owner: object
 ): maybeComponent is ComponentLike {
-  if (macroCondition(dependencySatisfies('ember-source', '>=3.25.0-beta.1'))) {
+  // SAFETY: in more recent versions of @glimmer/manager,
+  //         this throws an error when maybeComponent does not have
+  //         an associated manager.
+  try {
```
Comment (NullVoxPopuli) [inline addon/addon-test-support/@ember/test-helpers/-internal/is-component.ts:20]: It appears that `@glimmer/manager` changed its behavior. This change absorbs the change in `@glimmer/manager`

**Review by MelSumner — APPROVED** [review]
These changes seem correct to me, and I'll wait until tomorrow before merging in case anyone else wants to take a second look.

**Pull request merged**